pkg/services/alerting/notifiers/slack.go

```diff
@@ -115,7 +115,7 @@ func init() {
         <span class="gf-form-label width-8">Token</span>
         <input type="text"
           class="gf-form-input max-width-30"
-          ng-model="ctrl.model.settings.token"
+          ng-model="ctrl.model.secureSettings.token"
```

Comment:
Now when we have `secureFields` available in the UI (see earlier comment) we can render this with an unconfigured input field or a configured field:
![image](https://user-images.githubusercontent.com/1668778/85636251-418c3c00-b680-11ea-84bc-4cd72d931110.png)
![image](https://user-images.githubusercontent.com/1668778/85636242-3cc78800-b680-11ea-8fb1-e99a42679cfa.png)

Something along these lines should do this:
```html
<div class="gf-form max-width-30">
	<div class="gf-form gf-form--v-stretch"><label class="gf-form-label width-8">Token</label></div>
	<div class="gf-form gf-form--grow" ng-if="!ctrl.model.secureFields.token">
		<input type="text"
			class="gf-form-input max-width-30"
			ng-model="ctrl.model.secureSettings.token"
			data-placement="right">
		</input>
		<info-popover mode="right-absolute">
			Provide a bot token to use the Slack file.upload API (starts with "xoxb"). Specify Recipient for this to work
		</info-popover>
	</div>
	<div class="gf-form" ng-if="ctrl.model.secureFields.token">
		<input type="text" class="gf-form-input max-width-18" disabled="disabled" value="configured" />
		<a class="btn btn-secondary gf-form-btn" href="#" ng-click="ctrl.model.secureFields.token = false">reset</a>
	</div>
</div>
```

Reply:
I would suggest the `URL` field to be stored encrypted as well. What do you think?

Reply:
If URL is Incoming webhook, it should be encrypted. I add commit to encrypt URL.

Reply:
Great